teuthology/openstack/__init__.py

```diff
@@ -79,7 +79,7 @@ def __init__(self, name_or_id, info=None):
             self.set_info()
         else:
             self.info = dict(map(lambda (k,v): (k.lower(), v), info.iteritems()))
-        if self.info['status'] == 'ERROR':
+        if isinstance(self.info, dict) and self.info.get('status', '') == 'ERROR':
```

Comment:
I would prefer:
```
if self.info and self.info.get('status').lower() == 'error':
```
because, dict() always returns dict or None.

Reply:
TMTOWTODI

Reply:
I need to stop commenting...